janus_messages/src/time.rs

```diff
@@ -1,6 +1,6 @@
 //! Utilities for timestamps and durations.
```

Comment:
I think this module should continue to live in `janus_core`, for a couple of reasons:

* The intent of `janus_messages` is to be a message serialization/deserialization crate. Including a clock abstraction is outside this crate's wheelhouse.
* I am unsure _we_ want to use this `Clock` abstraction in the long run (#445), and therefore even less sure we want to export this as part of our public interface. (it already is as part of `janus_core`, natch, but anything in `janus_messages` is not just technically public but explictly moved to allow others to depend on it)

My suggestion would be to:
* Move this module back to `janus_core`.
* Remove any functionality in `janus_messages` that relies on a `Clock`.
* Replace `Nonce::generate` with `Nonce::new` that receives takes a `Time` instead of generating one from a `Clock`.
* Re-implement the removed `Nonce::generate` functionality in Janus (`janus_core`?), likely via an extension trait on `Nonce`.

This implementation strategy would be minimally impactful on existing callers to e.g. `Nonce::generate()`; any files that called `Nonce::generate` would need to import the new extension trait, but would otherwise be unchanged.